cmd/influxd/launcher/engine.go

```diff
@@ -0,0 +1,158 @@
+package launcher
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/http"
+	"github.com/influxdata/influxdb/kit/prom"
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/storage"
+	"github.com/influxdata/influxdb/storage/readservice"
+	"github.com/influxdata/influxdb/tsdb"
+	"github.com/influxdata/influxdb/tsdb/cursors"
+	"github.com/influxdata/influxql"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+)
+
+var _ Engine = (*storage.Engine)(nil)
+
+// Engine defines the time-series storage engine.  Wraps *storage.Engine
+// to facilitate testing.
+type Engine interface {
+	influxdb.DeleteService
+	readservice.View
+	storage.PointsWriter
+	storage.BucketDeleter
+	prom.PrometheusCollector
+
+	SeriesCardinality() int64
+
+	WithLogger(log *zap.Logger)
+	Open(context.Context) error
+	Close() error
+}
+
+var _ Engine = (*TemporaryEngine)(nil)
+var _ http.Flusher = (*TemporaryEngine)(nil)
+
+// TemporaryEngine creates a time-series storage engine backed
+// by a temporary directory that is removed on Close.
+type TemporaryEngine struct {
+	path    string
+	config  storage.Config
+	options []storage.Option
+
+	opened bool
+	engine *storage.Engine
+
+	logger *zap.Logger
+}
+
+// NewTemporaryEngine creates a new engine that places the storage engine files into
+// a temporary directory; used for testing.
+func NewTemporaryEngine(c storage.Config, options ...storage.Option) *TemporaryEngine {
+	return &TemporaryEngine{
+		config:  c,
+		options: options,
+		logger:  zap.NewNop(),
+	}
+}
+
+// Open creates a temporary directory and opens the engine.
+func (t *TemporaryEngine) Open(ctx context.Context) error {
+	if t.opened {
+		return nil
+	}
+
+	path, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+
+	t.path = path
+	t.engine = storage.NewEngine(path, t.config, t.options...)
+	t.engine.WithLogger(t.logger)
+
+	if err := t.engine.Open(ctx); err != nil {
+		_ = os.RemoveAll(path)
+		return err
+	}
+
+	t.opened = true
+	return nil
+}
+
+// Close will remove the directory containing the time-series files.
+func (t *TemporaryEngine) Close() error {
+	t.opened = false
```

Comment:
I imagine this will be ok as I imagine it to be used in fairly serial tests.  However, if someone needs to debug this code because I didn't protect it, they will hate me :)  I'm going to make it safer.

Reply:
(i ended up adding a mutex for just open/close)